raml-parser-interface/src/main/java/org/mule/apikit/common/ApiVendorUtils.java

```diff
@@ -21,8 +21,29 @@
 
 public class ApiVendorUtils {
 
-  private static final String OPENAPI_REGEX = "\\s*\"?\\s*openapi\\s*\"?\\s*:\\s*[\"|']?\\s*3\\.0.\\d+\\s*[\"']?\\s*.*$";
-  private static final String SWAGGER_REGEX = "\\s*\"?\\s*swagger\\s*\"?\\s*:\\s*[\"|']?\\s*2\\.0\\s*[\"']?\\s*.*$";
+  private static final String OPENAPI_REGEX;
+  private static final String SWAGGER_REGEX;
+  static {
+    // We want to match the something like openapi:3.0.<minor> or swagger:3.0.<minor> (with optional "" around either
+    // the spec type or the version).
+    //
+    // The version declaration might appear in an almost arbitrary place on the file (JSON or YAML). We want to match it
+    // nonetheless. The only circumstance where a match should be discarded is if it appears inside a comment. JSON has
+    // no syntax for comments, we just have to forbid YAML comments (they start on a # and end on a newline).
+    //
+    // We process the input line-by-line. It should be enough to assert that no # appears before the version
+    // declaration. This is an overly conservative approximation but should work for any non-adversarial input.
+    final String notAComment = "[^#]*";
+    final String maybeDoubleQuotes = "\"?";
+    final String maybeDoubleOrSingleQuotes = "[\"']?";
+    final String skipWhitespace = "\\s*";
+    final String matchPrefix = notAComment + skipWhitespace + maybeDoubleQuotes + skipWhitespace;
```

Comment:
Skipping whitespace here is not neccesary because we can already match that with `notAComment`.
```suggestion
    final String matchPrefix = notAComment + maybeDoubleQuotes + skipWhitespace;
```